input/otlp/logs.go

```diff
@@ -152,14 +154,19 @@ func (c *Consumer) convertLogRecord(
 		)
 	}
 
+	if eventDomain == "device" && eventName != "" {
+		event.Event.Category = "device"
+		event.Event.Kind = "event"
+		event.Event.Action = eventName
+	}
+
 	if eventName == "crash" {
 		if event.Error == nil {
 			event.Error = &model.Error{}
 		}
 		event.Error.Type = "crash"
-	}
-	if eventDomain == "device" {
-		event.Event.Category = "device"
+		event.Event.Kind = ""
+		event.Event.Action = ""
 	}
```

Comment:
Can you update this to follow https://github.com/elastic/apm-data/pull/28#issuecomment-1539268769?

> - if both `event.domain` and `event.name` are set, then we infer that it's an event log and set `event.kind: event`
> - for such event logs, if `event.domain: device` then we set `event.category: device`
> - for such event logs, we set `event.action: <event.name>`

Reply:
Sorry! My eyes were drawn to the changes in the crash code block and I didn't pay close attention to the other code block.

IIUC, "crash" events are only going to come from mobile devices, so I think we should combine them and avoid setting the Kind & Action fields only to clear them out afterwards.

```go
if eventDomain == "device" && eventName != "" {
	event.Event.Category = "device"
	if eventName == "crash" {
		if event.Error == nil {
			event.Error = &model.Error{}
		}
		event.Error.Type = "crash"
	} else {
		event.Event.Kind = "event"
		event.Event.Action = eventName
	}
}
```

WDYT?